src/potions.ts

```diff
@@ -211,7 +211,7 @@ export class Potion {
     }[] = [];
     const limitFunction = limit
       ? (quantity: number) =>
-          Math.min(limit - values.reduce((_total, tier) => tier.quantity, 0), quantity)
+          Math.min(limit - values.reduce((total, tier) => total + tier.quantity, 0), quantity)
```

Comment:
I don't off the top of my head get what this does, but I trust you

Reply:
i can rewrite it to use sumNumbers

Reply:
oh my god now I see why that's an issue

yeah go ahead and use sumNumbers just because

that's hilarious.